FFmpegInterop/SampleProvider.cpp

```diff
@@ -0,0 +1,414 @@
+//*****************************************************************************
+//
+//	Copyright 2015 Microsoft Corporation
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	http ://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+//
+//*****************************************************************************
+
+#include "pch.h"
+#include "SampleProvider.h"
+#include "Reader.h"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Media::Core;
+using namespace winrt::Windows::Media::MediaProperties;
+using namespace std;
+
+namespace winrt::FFmpegInterop::implementation
+{
+	SampleProvider::SampleProvider(_In_ AVStream* stream, _In_ Reader& reader) :
+		m_stream(stream),
+		m_reader(reader)
+	{
+		if (m_stream->start_time != AV_NOPTS_VALUE)
+		{
+			m_startOffset = m_stream->start_time;

```

Comment:
Wouldn't you need to set m_startOffset to the start_time in the AVFormatContext instead? Otherwise, you'd rebase all streams to start at 0 and re-introduce the bug which was fixed in b4a95dc.

Reply:
Yeah we need to use AVFormatContext::start_time here to avoid the desync issue.